lib/internal/Magento/Framework/Setup/Patch/PatchApplier.php

```diff
@@ -238,10 +237,9 @@ public function applySchemaPatch($moduleName = null)
                  * @var SchemaPatchInterface $schemaPatch
                  */
                 $schemaPatch = $this->patchFactory->create($schemaPatch, ['schemaSetup' => $this->schemaSetup]);
-                $schemaPatch->apply();
-                $this->patchHistory->fixPatch(get_class($schemaPatch));
-                foreach ($schemaPatch->getAliases() as $patchAlias) {
-                    $this->patchHistory->fixPatch($patchAlias);
+                if (!$this->checkPatchAliases($schemaPatch)) {
+                    $schemaPatch->apply();
+                    $this->patchHistory->fixPatch(get_class($schemaPatch));
```

Comment:
@Usik2203 We should still also loop over every alias and mark them as applied. In the case where a patch file is renamed (e.g. changing the namespace of a module), if we install the new version and only mark the new name as installed, rolling back to before the change will result in the same patch applying a second time. If we mark all aliases as installed, then it will prevent that from happening.